assets/js/components/LoginControl.js

```diff
@@ -0,0 +1,353 @@
+   *
+   * @param kebabCase the kebab-cased word (which is a data-attribute-name)
+   * @returns {(String|null)}
+   */
+  static convertKebabCaseToCamelCase(kebabCase = '') {
+    if (!kebabCase || typeof kebabCase !== 'string') {
+      return null;
+    }
+
+    return kebabCase.replace(/\-([a-z])/g, (x) => x.substring(1).toUpperCase());
+  }
+
+  /**
+   * Build the control
+   *
+   * This is the top level entry point for building the required DOM elements.
+   *
+   * @param {Array.<{text: String, uri: String}>} extraLinksToBuild data defining the links to build
+   * @param {HTMLElement} $elm pattern-root element
+   * @param {Function} buildElement Function used to build an element (elife-utils.buildElement)
+   * @return {HTMLElement} The resulting control element
+   */
+  buildControl(extraLinksToBuild, $elm, buildElement) {
+    const iconData = {
+      pathRoot: $elm.dataset.iconPathRoot,
+      srcset: $elm.dataset.iconSrcset,
+      altText: $elm.dataset.iconAltText
+    };
+
+    const $nav = buildElement.call(null, 'nav');
+    this.insertToggle($nav, iconData, buildElement);
+    this.$menu = this.buildMenu(extraLinksToBuild, buildElement);
+    $nav.appendChild(this.$menu);
+
+    return $nav;
+  }
+
+  /**
+   * Build and insert the menu toggle into the supplied container
+   *
+   * @param {HTMLElement} $container The required parent of the toggle
+   * @param {Object} iconData data describing the icon
+   * @param {String} iconData.pathRoot a common path component to all versions (so minus the file extension)
+   * @param {String} iconData.srcset the srcset to use on the <img>
+   * @param {String} iconData.atlText the text for the [alt]
+   * @param {Function} buildElement Function used to build an element (elife-utils.buildElement)
+   */
+  insertToggle($container, iconData, buildElement) {
+    const $toggle = buildElement.call(null, 'a', ['login-control__controls_toggle'], '', $container);
+    $toggle.setAttribute('href', '#');
+    $toggle.appendChild(LoginControl.buildIcon(iconData, buildElement));
+    $toggle.addEventListener('click', this.toggle.bind(this));
+  }
+
+  /**
+   * @param {Object} iconData data describing the icon
+   * @param {String} iconData.pathRoot a common path component to all versions (so minus the file extension)
+   * @param {String} iconData.srcset the srcset to use on the <img>
+   * @param {String} iconData.atlText the text for the [alt]
+   * @param {Function} buildElement Function used to build an element (elife-utils.buildElement)
+   * @returns {HTMLElement} the <picture> describing the icon
+   */
+  static buildIcon(iconData, buildElement) {
+    if (!(iconData.pathRoot && iconData.srcset && iconData.altText)) {
+      return null;
+    }
+
+    const $icon = buildElement.call(null, 'picture');
+    const $source = buildElement.call(null, 'source', [], '', $icon);
+    $source.setAttribute('srcset', `${iconData.pathRoot}.svg`);
```

Comment:
Journal should just provide a complete URI. User-loaded avatars won't (probably) be SVG.